### PR TITLE
Freezing deps for 1.8 release

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

- Freezes deps versions for 1.8 release
- Updates CI

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -2,35 +2,49 @@ name: CI
 
 on:
   push:
+  schedule:
+    - cron: "0 0 * * *"
```
Comment (sanchezfdezjavier) [inline .github/workflows/main.yml:6]: Do we need to run this daily?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -58,3 +78,20 @@ jobs:
...
+    runs-on: ubuntu-20.04
+    name: Charmhub upload
+    needs:
+      - unit-tests-with-coverage
+      - integration-test
+    if: github.ref_name == '1.8'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@2.2.2
```
Comment (sanchezfdezjavier) [inline .github/workflows/main.yml:92]: Version `2.2.5` is out, any reason why we're not freezing to the latest?

File: .github/workflows/main.yml (hunk 1 above)
Comment (Gmerold) [reply .github/workflows/main.yml:6]: We don't need, but we want to. Release branch is unlikely to have a lot of changes, so we want to run the CI periodically, to make sure it's still working.

File: test-requirements.txt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+git+https://github.com/juju/python-libjuju.git
```
Comment (sanchezfdezjavier) [inline test-requirements.txt:1]: Is there any reason why we're leaving this to fetch from the upstream?

File: .github/workflows/main.yml (hunk 2 above)
Comment (Gmerold) [reply .github/workflows/main.yml]: Updated.

File: .github/workflows/promote.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            echo "promote-to=1.8/candidate" >> ${GITHUB_ENV}
+          elif [ "${PROMOTION}" == "1.8/candidate -> 1.8/stable" ]; then
+            echo "promote-from=1.8/candidate" >> ${GITHUB_ENV}
+            echo "promote-to=1.8/stable" >> ${GITHUB_ENV}
+          elif [ "${PROMOTION}" == "1.8/stable -> latest/stable" ]; then
+            echo "promote-from=1.8/stable" >> ${GITHUB_ENV}
+            echo "promote-to=latest/stable" >> ${GITHUB_ENV}
+          fi
+      - name: Promote Charm
+        uses: canonical/charming-actions/release-charm@2.2.2
```
[the remark this answers is not in the document]
Comment (Gmerold) [reply .github/workflows/promote.yml]: Updated.

File: test-requirements.txt (hunk 3 above)
Comment (sanchezfdezjavier) [reply test-requirements.txt:1]: And same for the package below

Comment (Gmerold) [reply test-requirements.txt]: In fact, I'm not even sure how it ended up here. I thought I removed it. 
It's gone now.

**Review by sanchezfdezjavier — APPROVED** [review]

File: test-requirements.txt (hunk 3 above)
Comment (sanchezfdezjavier) [reply test-requirements.txt]: Happens to me frequently haha

**Review by ghislainbourgeois — APPROVED** [review]

**Review by saltiyazan — APPROVED** [review]

File: tox.ini [hunk 5]
```diff
@@ -8,13 +8,12 @@ envlist = lint, static, unit
...
 src_path = {toxinidir}/src/
 unit_test_path = {toxinidir}/tests/unit/
-integration_test_path = {toxinidir}/tests/integration/
-all_path = {[vars]src_path} {[vars]unit_test_path} {[vars]integration_test_path}
+all_path = {[vars]src_path} {[vars]unit_test_path}
 
 [testenv]
 deps =
-    pytest
-    pytest-operator
```
Comment (saltiyazan) [inline tox.ini:17]: isn't `pytest-operator` needed?

**Review by gruyaume — APPROVED** [review]

**Pull request merged**